### ISSUE TITLE
License type

### ISSUE THREAD

**Issue by Nekto89** [KarypisLab/GKlib#30]
This library uses several copy-pastes of code under LGPL license. Do I understand correctly that it can't be used under just Apache 2.0 license terms?

**Comment by Nekto89** [comment]
> This would be quite serious issue to address if this project is to be packaged in downstream. Do you remember what parts were copied from where?

https://github.com/KarypisLab/GKlib/blob/6e7951358fd896e2abed7887196b6871aac9f2f8/include/gk_mksort.h#L22C37-L23C11
https://github.com/KarypisLab/GKlib/blob/6e7951358fd896e2abed7887196b6871aac9f2f8/src/getopt.c#L19
https://github.com/KarypisLab/GKlib/blob/6e7951358fd896e2abed7887196b6871aac9f2f8/include/gkregex.h#L8C37-L9C11
https://github.com/KarypisLab/GKlib/blob/6e7951358fd896e2abed7887196b6871aac9f2f8/src/gkregex.c#L7C37-L8C11

**Comment by karypis** [comment]
Resolved by #62 (merged as 9725f79). Rather than relicense, the mixed license is now documented per @LecrisUT's suggestion:

- \`LICENSES.md\` inventories every non-Apache file: the four LGPL-2.1-or-later files derived from the GNU C Library (\`include/gk_mksort.h\`, \`include/gkregex.h\`, \`src/getopt.c\`, \`src/gkregex.c\`), plus one the thread had not noted — the MT19937 generator in \`src/random.c\`, which is BSD-3-Clause.
- Full license texts are shipped under \`LICENSES/\` (\`LGPL-2.1.txt\`, \`BSD-3-Clause-MT.txt\`).
- \`LICENSE.txt\` and \`README.md\` now state the effective combined license: \`Apache-2.0 AND LGPL-2.1-or-later AND BSD-3-Clause\`.

This gives downstream packagers an SPDX-style expression and per-file provenance. No code was relicensed or removed; swapping the LGPL components for permissive equivalents remains a possible future cleanup if a fully-Apache build is ever desired.

**Comment by gfaster** [comment]
I do not believe LGPL allows including source files and just declaring multiple licenses. GKLib would be classified as a "work based on the library" and thus according to the terms must be released under the LGPL. I would expect to see such a change from the machine that appears to now speak for the original author of GKlib.

**Comment by LecrisUT** [comment]
It didn't take me long to find an issue with the PR. I am not comfortable on how this was handled without a proper review, instead it was just fed into LLM which will always do things very literally, but rarely correct
